include/itkMultiScaleHessianEnhancementImageFilter.hxx

```diff
@@ -0,0 +1,248 @@
+    progress->RegisterInternalFilter(m_EigenToScalarImageFilter, perFilterProccessPercentage);
+
+    /* Check if we need to run the MaximumAbsoluteValueFilter at all */ 
+    if (m_SigmaArray.GetSize() > 1)
+    {
+      progress->RegisterInternalFilter(m_MaximumAbsoluteValueFilter, perFilterProccessPercentage);
+    }
+    else
+    {
+      itkDebugMacro(<< "maximumAbsoluteValueFilter is not being used");
+    }
+  }
+
+  /* We store a single pointer that we will graft to the output */
+  typename TOutputImage::Pointer outputImageTypePointer;
+
+  /* Process the first scale */
+  outputImageTypePointer = generateResponseAtScale((SigmaType)0);
+
+  /* Process the remaining sigma values */
+  for (SigmaStepsType scaleLevel = 1; scaleLevel < m_SigmaArray.GetSize(); ++scaleLevel)
+  {
+    /* Calculate next response value */
+    typename TOutputImage::Pointer tempResponseOutputImageTypePointer = generateResponseAtScale(scaleLevel);
+
+    /* Take absolute value maximum */
+    m_MaximumAbsoluteValueFilter->SetInput1(outputImageTypePointer);
+    m_MaximumAbsoluteValueFilter->SetInput2(tempResponseOutputImageTypePointer);
+    m_MaximumAbsoluteValueFilter->Update();
+
+    /* Save max and go to next sigma value */
+    outputImageTypePointer = m_MaximumAbsoluteValueFilter->GetOutput();
+  }
+
+  /* Graft output and we're done! */
+  this->GraftOutput(outputImageTypePointer);
+}
+
+template< typename TInputImage, typename TOutputImage >
+typename TOutputImage::Pointer
+MultiScaleHessianEnhancementImageFilter< TInputImage, TOutputImage >
+::generateResponseAtScale(SigmaStepsType scaleLevel)
+{
+  /* Get this sigma value */
+  SigmaType thisSigma = m_SigmaArray.GetElement(scaleLevel);
+
+  /* Process pipeline and return */
+  m_HessianFilter->SetSigma(thisSigma);
+  m_EigenToScalarImageFilter->Update();
+  return m_EigenToScalarImageFilter->GetOutput();
+}
+
+template< typename TInputImage, typename TOutputImage >
+typename MultiScaleHessianEnhancementImageFilter< TInputImage, TOutputImage >::SigmaArrayType
+MultiScaleHessianEnhancementImageFilter< TInputImage, TOutputImage >
+::GenerateSigmaArray(SigmaType SigmaMinimum, SigmaType SigmaMaximum, SigmaStepsType NumberOfSigmaSteps, SigmaStepMethodEnum SigmaStepMethod)
+{
+  /* Quick check to make sure value is correct */
+  if ( NumberOfSigmaSteps < 1 )
+  {
+    /*
+    * Since this is a static class, we have no "this" pointer so we cannot throw an exception. Instead
+    * we will return an empty sigma array.
+    */
+    // itkExceptionMacro(<< "NumberOfSigmaSteps (" << NumberOfSigmaSteps << ") must be one or greater");
+    SigmaArrayType sigmaArrayEmpty;
+    sigmaArrayEmpty.SetSize(0);
+    return sigmaArrayEmpty;
+  }
+
+  /* If we have min greater than max just swap */
+  if (SigmaMinimum > SigmaMaximum) {
+    std::swap(SigmaMinimum, SigmaMaximum);
+  }
+
+  /* If min equals max, just generate one step size */
+  if (SigmaMinimum == SigmaMaximum) {
+    NumberOfSigmaSteps = 1;
```

Comment:
Leads to a division by 0 further down

Reply:
Good find. Strange that it passed the tests... See ce229e2d94ba40f90588298a8e0db649bf84e491